src/java/org/apache/cassandra/db/compaction/LeveledCompactionStrategy.java

```diff
@@ -372,6 +380,16 @@ protected OnDiskAtomIterator computeNext()
             }
         }
 
+        private Pair<String, String> sstablesToKsCfPair(Collection<SSTableReader> sstableReaders)
+        {
+            Set<Pair<String, String>> ksCfPairs = sstableReaders.stream().map(sstable -> Pair.create(sstable.getKeyspaceName(), sstable.getColumnFamilyName()))
+                                                                         .collect(Collectors.toSet());
+            Preconditions.checkArgument(ksCfPairs.size() == 1,
+                                        "An individual compaction should span a single keyspace/table pair {}",
+                                        SafeArg.of("keyspaceTablePair", ksCfPairs));
+            return ksCfPairs.stream().findFirst().get();
+        }
+
         public void close() throws IOException
         {
             if (currentScanner != null)
```

Comment:
is there any reason you didn't also apply the changes to `AbstractCompactionStrategy`?

Reply:
We don't use STCS / DTCS to any capacity, beyond that no. Happy to put it in if its desirable.

Reply:
how hard do you think it would be to add to the non-LCS ones (presumably just to `AbstractCompactionStrategy`)? i'd prefer to be consistent throughout and add them in this PR, but defer to you on what you want to do here

Reply:
Not hard. It _can_ be a one liner but I've refactored the `sstablesToKsCfPair` to live in ACS.